spec/index.bs

```diff
@@ -642,36 +682,79 @@ interface GPUAdapter {
     **Arguments:**
         - optional {{GPUDeviceDescriptor}} |descriptor| = {}
 
-    **Returns:** |promise|, of type Promise<{{GPUDevice}}>
+    **Returns:** |promise|, of type Promise<{{GPUDevice}}>.
 
     Requests a [=device=] from the [=adapter=].
 
-    Returns [=a new promise=], |promise|.
-
-      - If the following requirements are met, |promise| [=resolves=] to a new
-        {{GPUDevice}} object encapsulating a new [=device=], which MUST
-        have exactly the capabilities described by |descriptor|.
+    Returns [=a new promise=], |promise|. Then, the following occurs asynchronously:
 
-        Issue: Figure out how to integrate the requirements.
+      - If the user agent can fulfill the request and
+        the [$requestDevice Valid Usage|valid usage$] rules are met,
+        |promise| [=resolves=] to a new {{GPUDevice}} object encapsulating
+        [=a new device=] with the capabilities described by |descriptor|.
 
       - Otherwise, |promise| [=rejects=] with an {{OperationError}}.
 </div>
 
+<div algorithm class=validusage>
+    <dfn abstract-op>requestDevice Valid Usage</dfn>
+
+    Given an [=adapter=] |adapter| and a {{GPUDeviceDescriptor}} |descriptor|,
+    the following validation rules apply:
+
+      - The set of {{GPUExtensionName}} values in
+        |descriptor|.{{GPUDeviceDescriptor/extensions}}
+        must be a subset of those in |adapter|.{{adapter/[[extensions]]}}.
+
+      - For each type of limit in {{GPULimits}}, the value of that limit in
+        |descriptor|.{{GPUDeviceDescriptor/limits}}
+        must be less than or equal to the value of that limit in
```

Comment:
ditto, limits are not always maximums. We could use the word "better" and define for each limits what better means.

Reply:
how about "must not exceed the corresponding limit in adapter.limits" or something similar?

Reply:
if it's an alignment, than "exceed" becomes backwards